mail/local/mailbox.go

```diff
@@ -0,0 +1,256 @@
+
+		// Start listening for remote changes
+		events, err := m.clients.Threads.Listen(ctx, boxID, []tc.ListenOption{{
+			Type:       tc.ListenAll,
+			Collection: mail.CollectionName,
+		}})
+		if err != nil {
+			state <- cmd.WatchState{Err: err, Fatal: !cmd.IsConnectionError(err)}
+			return
+		}
+		errs := make(chan error)
+		go func() {
+			for e := range events {
+				if e.Err != nil {
+					errs <- e.Err // events will close on error
+					continue
+				}
+				switch e.Action.Type {
+				case tc.ActionCreate, tc.ActionSave:
+					var msg client.Message
+					if err := msg.Unmarshal(e.Action.Instance); err != nil {
+						errs <- err
+						return
+					}
+					var t MailboxEventType
+					if e.Action.Type == tc.ActionCreate {
+						t = NewMessage
+					} else {
+						t = MessageRead
+					}
+					mevents <- MailboxEvent{
+						Type:      t,
+						MessageID: db.InstanceID(e.Action.InstanceID),
+						Message:   msg,
+					}
+				case tc.ActionDelete:
+					mevents <- MailboxEvent{
+						Type:      MessageDeleted,
+						MessageID: db.InstanceID(e.Action.InstanceID),
+					}
+				}
+			}
+		}()
+
+		// If we made it here, we must be online
+		state <- cmd.WatchState{State: cmd.Online}
+
+		for {
+			select {
+			case err := <-errs:
+				state <- cmd.WatchState{Err: err, Fatal: !cmd.IsConnectionError(err)}
+				return
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+	return state, nil
+}
+
+// ReadInboxMessage marks a message as read by ID.
+func (m *Mailbox) ReadInboxMessage(ctx context.Context, id string) error {
+	ctx, err := m.context(ctx)
+	if err != nil {
+		return err
+	}
+	return m.clients.Users.ReadInboxMessage(ctx, id)
+}
+
+// DeleteInboxMessage deletes an inbox message by ID.
+func (m *Mailbox) DeleteInboxMessage(ctx context.Context, id string) error {
+	ctx, err := m.context(ctx)
+	if err != nil {
+		return err
+	}
+	return m.clients.Users.DeleteInboxMessage(ctx, id)
+}
+
+// DeleteSentMessage deletes a sent message by ID.
+func (m *Mailbox) DeleteSentMessage(ctx context.Context, id string) error {
+	ctx, err := m.context(ctx)
+	if err != nil {
+		return err
+	}
+	return m.clients.Users.DeleteSentMessage(ctx, id)
+}
+
+// Identity returns the mailbox's user identity.
+func (m *Mailbox) loadIdentity() error {
+	ids := m.conf.Viper.GetString("identity")
+	if ids == "" {
+		return fmt.Errorf("identity not found")
+	}
+	idb, err := base64.StdEncoding.DecodeString(ids)
+	if err != nil {
+		return fmt.Errorf("loading identity: %v", err)
+	}
+	m.id = &thread.Libp2pIdentity{}
+	if err = m.id.UnmarshalBinary(idb); err != nil {
+		return fmt.Errorf("unmarshalling identity: %v", err)
+	}
+	return nil
+}
+
+func (m *Mailbox) context(ctx context.Context) (context.Context, error) {
+	ctx = common.NewAPIKeyContext(ctx, m.conf.Viper.GetString("api_key"))
+	secret := m.conf.Viper.GetString("api_secret")
+	if secret != "" {
+		var err error
+		ctx, err = common.CreateAPISigContext(ctx, time.Now().Add(time.Hour), secret)
```

Comment:
Q: So if someone is watching (in/sent)boxes for more than an hour, the watching logic might fail by an expired signature, right? 
I think that deadline is OK, mostly to validate understanding in that situation. We can always imagine handling that error in the future and auto-refresh sign.

Reply:
Ah, good catch! I'll create an issue.